### DISCUSSION TITLE
ExplorerPatcher crash when changing language

### DISCUSSION THREAD

**Discussion by kern242** [valinet/ExplorerPatcher#538] [Q&A]
ExplorerPatcher crash every time when changing language by press on shift+alt
please take look on it hope you can fix it,
big thanks on your awesome work

https://user-images.githubusercontent.com/95850880/145372293-b4784de1-fec1-49c0-b7b3-5b69bf0beb51.mp4

**Comment by BraINstinct0** [comment]
I am not really sure what I'm seeing there...seems like you've moved the Start menu to the right manually or was it like that from the start?

**Reply by valinet** [reply]
It’s right to left layout, the OS is probably set to a language that’s written from right to left, doesn’t really matter, it’s not because of that, or it shouldn’t be.

(As an aside, in an RTL language, the window buttons are placed on the left of the window, so this means there is some code path and theoretically a way to hack dwm and have it display the window buttons on the left, which would be very macOS-like and interesting, I think of this from time to time, never took the time to actually try it).

**Comment by valinet** [comment]
Well, I don’t know what it could be. Does it happen with Windows+Space as well? What Windows build? What EP version? What window switcher are you using? Try collecting a crash dump: https://docs.microsoft.com/en-us/windows/win32/wer/collecting-user-mode-dumps and maybe we find the answer in there, if it’s EP to blame. First, this should ideally be reproduced on a clean VM, to make sure it’s nothing else that interferes. I cannot figure it out myself from a video alone, I in no way am no wizard.

**Reply by kern242** [reply]
yes it happen with Windows+Space as well
Windows build: Windows 11 Pro Insider Preview 22518.1000
EP version: 22000.348.40.6 
just upload the crash dump: explorer.exe.12744.dmp.txt
[explorer.exe.12744.txt](https://github.com/valinet/ExplorerPatcher/files/7685924/explorer.exe.12744.txt)

**Reply by valinet** [reply]
What window switcher do you use? Does this happen on stable version as well (39.2)?

**Reply by valinet** [reply]
Also, what language switcher have you configured in settings? The problem seems to be that failing to initialize for whatever reason.

**Reply by kern242** [reply]
language switcher: hebrew
i cannot confirm it happen to stable version i have in my posses only this main computer
window switcher do you mean shift+alt keys ?

**Comment by Galixte** [comment]
> > “…Does this happen on stable version as well (39.2)?”
> i cannot confirm it happen to stable version i have in my posses only this main computer

@valinet talks about the latest stable version of ExplorerPatcher available here: https://github.com/valinet/ExplorerPatcher/releases/tag/22000.348.39.1_493cf27

> > “What window switcher do you use?…”
> window switcher do you mean shift+alt keys ?

“Window switcher” within ExplorerPatcher settings, such as these:
![Windows switcher process](https://user-images.githubusercontent.com/7478287/145471640-887e4457-7b0a-4d18-9b07-edf60d21ff81.png)

**Reply by kern242** [reply]
![Desktop Screenshot 2021 12 09 - 23 02 27 67](https://user-images.githubusercontent.com/95850880/145475593-ac105525-e0d3-41bc-aa71-86cdf3e15407.png)
![Desktop Screenshot 2021 12 09 - 23 04 19 23](https://user-images.githubusercontent.com/95850880/145475595-5cff8499-d56b-4512-bf61-7abeb6a5b220.png)

yes same problem also in the latest stable version of ExplorerPatcher 
i just uploaded screenshot of ExplorerPatcher window switcher and taskbar settings
*when i change back the taskbar to windows 11 instead of 10 the problem is vanished but then the windows 11 taskbar is annoying.

**Reply by valinet** [reply]
Which language switcher? Please upload a screenshot of the "System tray" section as well.

**Reply by kern242** [reply]
@valinet
thank you very much it was set by default on windows 11 i just change it to windows 10
and restart file explorer and the problem solved Big Thank you
![Desktop Screenshot 2021 12 09 - 23 29 58 61](https://user-images.githubusercontent.com/95850880/145478776-000e2072-2227-4432-ae34-690553d54d55.png)

**Reply by valinet** [reply]
Yeah, well, it's weird, like, it should work when set to Windows 11 as well, like, that's the default =))) It works on some configurations, I can use it on my PC just fine, for example. So the new Windows 11 one fails to initialize for you. Idk why, it's weird, when set to that, I don't mess with it directly, so idk. At least there is a workaround for the moment, you're welcome.

**Comment by Andre-10** [comment]
Can confirm this. I'm on Insider Dev version 22518.1000, and EP ver 39.1 (stable). 

Changing language (Win 11 style will crash Explorer. Changing it to Win 10 style it works fine
![image](https://user-images.githubusercontent.com/93877359/145643725-2817ff95-fbde-41a9-92ba-e2b61b3bcc85.png)

Forgot about language settings: Eng (US) and Slovenia. Both using Slovenian keyboard
